client/src/app/core/repositories/management/meeting-repository.service.ts

```diff
@@ -163,21 +169,56 @@ export class MeetingRepositoryService extends BaseRepository<ViewMeeting, Meetin
             }
         ];
 
-        if (addedUserIds?.length || removedUserIds?.length) {
-            if (!meeting.default_group_id) {
-                throw new Error('The default group is needed!');
-            }
+        let { removedUserIds = [] }: MeetingUserModifiedFields = options;
+        const { addedUserIds, addedAdminIds, removedAdminIds = [] }: MeetingUserModifiedFields = options;
+        // Users, removed from the default group, are not removed from the admin group
+        removedUserIds = removedUserIds.difference(addedAdminIds).difference(meeting.admin_group.user_ids);
 
-            const addData = (addedUserIds || []).map(id => ({
-                id,
-                group_$_ids: { [meeting.id]: [meeting.default_group_id] }
-            }));
-            const removeData = (removedUserIds || []).map(id => ({ id, group_$_ids: { [meeting.id]: [] } }));
+        const updateData: { id: Id; group_$_ids: { [meetingId: number]: Id[] } }[] = [];
+        // Adding a user as admin also adds they to the default-group
+        if (addedAdminIds?.length) {
+            updateData.push(...this.getUserGroupUpdatePayload(addedAdminIds, meeting.id, meeting.admin_group_id));
+        }
+        if (addedUserIds?.length) {
+            updateData.push(
+                ...this.getUserGroupUpdatePayload(
+                    addedUserIds.difference(addedAdminIds),
+                    meeting.id,
+                    meeting.default_group_id
+                )
+            );
+        }
+        if (removedAdminIds?.length) {
+            // Users, removed from the admin group, are not removed from the default group.
+            // But if they are removed from the whole meeting, remove them completely.
+            const remainingRemovedAdminIds = removedAdminIds.filter(
+                userId =>
+                    (meeting.user_ids.includes(userId) || addedUserIds.includes(userId)) &&
+                    !removedUserIds.includes(userId)
+            );
+            // If users removed from the default group but added to the admin group of a meeting, then they are still
+            // admins and can only be removed this way
+            const completelyRemovedAdminIds = removedAdminIds.filter(
+                userId =>
+                    !meeting.user_ids.includes(userId) &&
+                    !addedUserIds.includes(userId) &&
+                    !removedUserIds.includes(userId)
+            );
+            updateData.push(
+                ...this.getUserGroupUpdatePayload(remainingRemovedAdminIds, meeting.id, meeting.default_group_id)
+            );
+            updateData.push(...this.getUserGroupUpdatePayload(completelyRemovedAdminIds, meeting.id));
+        }
+        if (removedUserIds?.length) {
+            updateData.push(...this.getUserGroupUpdatePayload(removedUserIds, meeting.id));
+        }
```

Comment:
This is not so beautiful I wanted, but for now it fulfills its purpose and I will clean up it later.